src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,98 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task from the TaskList.
+     * @param index the index of task to delete
+     */
+    public void delete(int index) {
+        try {
+            Task t = tasks.remove(index - 1);
+            int length = tasks.size();
+            String output = length == 1 ? " task in the list." : " tasks in the list.";
+            System.out.println("Got it. I've deleted this task:\n" + t
+                    + "\nNow you have " + length + output);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Can consider encapsulating index out of bounds as a Duke exception here, and just trigger the Duke exception instead of sout the string message? But nonetheless, great effort in being so meticulous with all the possible error cases!